addons/pos_restaurant/static/src/overrides/models/pos_order.js

```diff
@@ -17,6 +17,9 @@ patch(PosOrder.prototype, {
     getTable() {
         return this.table_id;
     },
+    getOrderName() {
+        return this.table_id?.table_number.toString() || this.getFloatingOrderName() || "";
+    },
```

Comment:
inline this code in `split_bill_screen.js`. It looks like it's only needed there.